### PR TITLE
Put limit and offset normalization under test.

### PR DESCRIPTION
**Pull request by rgwozdz**

### CODE REVIEW

File: lib/normalize-query-options/limit.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Normalize the limit option; defaults to undefined
+ * @param {object} options
+ * @returns {integer} or undefined
+ */
+function normalizeLimit (options) {
+  const limit = options.limit || options.resultRecordCount || options.count || options.maxFeatures
```
Comment (haoliangyu) [inline lib/normalize-query-options/limit.js:7]: Can the user set limit=0?

File: lib/normalize-query-options/offset.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Normalize the offset option. If no limit is defined, then return offset as undefined. ala-sql
+ * requires OFFSET to be paired with a LIMIT
+ * @param {object} options
+ * @returns {integer} or undefined
+ */
+function normalizeOffset (options) {
+  return (options.limit) ? (options.offset || options.resultOffset) : undefined
```
Comment (haoliangyu) [inline lib/normalize-query-options/offset.js:8]: same for limit=0?

Comment (rgwozdz) [reply lib/normalize-query-options/offset.js:8]: Good catch.  This is the old code and just putting under test, but I'll make an adjustment to allow for 0 limit.

**Comment by rgwozdz** [comment]
@haoliangyu - handling limit with zero value. Offset value that has a value of zero, is the same as being undefined, so no changes there.

**Pull request merged**